includes/class-wc-stripe-intent-controller.php

```diff
@@ -1037,7 +1039,57 @@ public function create_and_confirm_setup_intent_ajax() {
 			wp_send_json_error(
 				[
 					'error' => [
-						'message' => $e->getLocalizedMessage(),
+						'message' => $e->getLocalizesdMessage(),
+					],
+				]
+			);
+		}
+	}
+
+	/**
+	 * Confirms the change payment method request for a subscription.
+	 *
+	 * This creates the payment method token from the setup intent.
+	 *
+	 * This function is used to confirm the change payment method request for a subscription after the user has been asked to authenticate their payment (eg 3D-Secure).
+	 * It is initiated from the subscription change payment method page.
+	 */
+	public function confirm_change_payment_from_setup_intent_ajax() {
+		try {
+			$is_nonce_valid = check_ajax_referer( 'wc_stripe_update_order_status_nonce', false, false );
+
+			if ( ! $is_nonce_valid ) {
+				throw new WC_Stripe_Exception( 'missing-nonce', __( 'CSRF verification failed.', 'woocommerce-gateway-stripe' ) );
+			}
+
+			if ( ! function_exists( 'wcs_is_subscription' ) ) {
+				throw new WC_Stripe_Exception( 'subscriptions_not_found', __( "We're not able to process this subscription change payment request payment. Please try again later.", 'woocommerce-gateway-stripe' ) );
+			}
+
+			$subscription_id = isset( $_POST['order_id'] ) ? absint( $_POST['order_id'] ) : false;
+			$subscription    = wcs_get_subscription( $subscription_id );
+
+			if ( ! $subscription ) {
+				throw new WC_Stripe_Exception( 'subscription_not_found', __( "We're not able to process this subscription change payment request payment. Please try again later.", 'woocommerce-gateway-stripe' ) );
+			}
+
+			$setup_intent_id = isset( $_POST['intent_id'] ) ? wc_clean( wp_unslash( $_POST['intent_id'] ) ) : null;
+
+			$gateway = $this->get_upe_gateway();
+			$gateway->create_token_from_setup_intent( $setup_intent_id, $subscription->get_user() );
```

Comment:
Same case here for `$setup_intent_id`. The first param of`create_token_from_setup_intent` is not type-strict, but a bug may happen with the Stripe request (due the generated URL path):
```
			$setup_intent = $this->stripe_request( 'setup_intents/' . $setup_intent_id . '?&expand[]=latest_attempt' );
```

Reply:
👍 Fixed in: [6b271c9](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3313/commits/6b271c975dda085fb15586379a8643bdc0787efc)